File: algorithms/sort/bubble.cpp
#include<bits/stdc++.h>
using namespace std;

void bubble_sort(vector<int> &arr){
  int n = arr.size();
  for (int i = 0; i < n-1; i++){
    for (int j = 0; j < n-i-1; j++){
      if(arr[j] > arr[j+1]) swap(arr[j], arr[j+1]);
    }    
  }
  
}

int main(){
  vector<int> arr = {4, 5, 1, 3, 2};
  bubble_sort(arr);
  for (int i = 0; i < arr.size(); i++){
    cout << arr[i] << " ";
  }  
} 

File: algorithms/sort/bubble.py
def bubble_sort(list):
  for i in range(len(list)-1):
    for j in range(len(list)-i-1):
      if list[j] > list[j+1]:
        list[j], list[j+1] = list[j+1], list[j]

list = [4, 5, 9, 8, 7, 3, 2, 1, 6]
bubble_sort(list)
print(list)
